src/ResourceManager/Insights/Commands.Insights/Diagnostics/RemoveAzureRmDiagnosticSettingCommand.cs

```diff
@@ -0,0 +1,149 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Management.Monitor;
+using Microsoft.Azure.Management.Monitor.Models;
+using Microsoft.Rest;
+using Microsoft.Rest.Azure;
+using System.Globalization;
+using System.Management.Automation;
+using System.Net;
+using System.Threading;
+
+namespace Microsoft.Azure.Commands.Insights.Diagnostics
+{
+    /// <summary>
+    /// Removes a named diagnostic setting or disables the setting called 'service' if the name argument is not present or if is 'service'.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzureRmDiagnosticSetting", SupportsShouldProcess = true), OutputType(typeof(AzureOperationResponse))]
```

Comment:
@pkamath13 @gucalder is this a long-running operation? If so, it would make sense to let the cmdlet wait on the operations success, and then return the corresponding object. Then we could add the `-AsJob` parameter to allow users to run the operation in the background of the PowerShell session.

Reply:
We currently have no long-running operations in our API.